### PR TITLE
(HAProxy) Update Haproxy chart to 1.16.1

### PR DESCRIPTION
**Pull request by andrewheberle**
This PR updates the HAproxy Helm chart version to 1.16.1 which includes version 1.6.5 of the HAProxy Kubernetes Ingress Controller.

### CODE REVIEW

File: packages/haproxy/generated-changes/patch/Chart.yaml.patch
Comment (andrewheberle) [reply packages/haproxy/generated-changes/patch/Chart.yaml.patch:14]: The display-name annotation has been added and the upstream chart already includes a kubeVersion value

**Comment by PennyScissors** [comment]
Could you please drop the following commits, then run a new `make charts` and commit in a new last commit?
1. Add HAProxy 1.16.1 built assets `614d614` (first `make charts` you ran)
2. Update generated chart `0d72727` (second `make charts` you ran)

The current behavior in our scripts has a bug that won't reconstruct the index based on the latest annotations so your display name annotation update is not being added to the index and therefore missing from the UI.

**Review by PennyScissors — CHANGES_REQUESTED** [review]
@andrewheberle could you please drop the following commits, then run a new `make charts` and commit assets/ charts/ and index.yaml in a new last commit?
1. Add HAProxy 1.16.1 chart build c635e8f (first `make charts` you ran)
2. Add HAProxy 1.16.1 built assets 614d614 (first `make charts` you ran)
3. Update generated chart 0d72727 (second `make charts` you ran)

The current behavior in our scripts has a bug that won't reconstruct the index based on the latest annotations so your display name annotation update is not being added to the index and therefore missing from the UI.

**Comment by andrewheberle** [comment]
@PennyScissors those commits have been dropped so there is now just the two in my branch, so hopefully that's all good to go.

**Comment by PennyScissors** [comment]
Hello @andrewheberle, could you please drop the last make charts commit, re-run make charts, and create a new commit once again? We recently merged a fix for our scripts that affected existing PRs and chart will need to be regenerated. Thanks! The PR looks good, so it should be ready to approve after.

**Comment by andrewheberle** [comment]
Commit `8e21fc6` has been dropped and new `make charts` added as commit `06b2a87`, however this failed CI checks.

Should I rebase my branch back onto `main-source` and start from scratch?

**Comment by PennyScissors** [comment]
@andrewheberle Before you try rebasing and doing it from scratch, try dropping the commit again, then run `make pull-scripts` and then `make charts`. It should generate a chart in assets named `haproxy-1.16.100.tgz` rather than `haproxy-1.16.100+up1.16.1.tgz` since we disabled appending the build metadata.

**Comment by andrewheberle** [comment]
I initially re-did the process with the `make pull-scripts` however that still pulled down v0.2.1 of the build scripts so I went ahead and rebased my branch off "main-source" and started from scratch so I ended up with the v0.2.2 build scripts.

Hopefully 5th time lucky :)

**Review by PennyScissors — APPROVED** [review]
Looks good now! Thank you, I appreciate your patience.

FYI, we switched to doing batch releases so this will not be released on merge. I currently have a PR open to release a few other partner charts so I will add yours to it to hopefully get this released sooner. Here is the [PR](https://github.com/rancher/partner-charts/pull/143) if you want to keep track of it.

**Pull request merged**
